Upstream: Yes, backport
Reason: Fixes a binary not being PIE while being SUID

See: https://github.com/hpcng/singularity/issues/5346

diff --git a/mlocal/frags/go_common_opts.mk b/mlocal/frags/go_common_opts.mk
index c1d3cb4..2f50f60 100644
--- a/mlocal/frags/go_common_opts.mk
+++ b/mlocal/frags/go_common_opts.mk
@@ -3,7 +3,7 @@ GO111MODULE := on
 GO_TAGS := containers_image_openpgp sylog imgbuild_engine oci_engine singularity_engine fakeroot_engine
 GO_TAGS_SUID := containers_image_openpgp sylog singularity_engine fakeroot_engine
 GO_LDFLAGS :=
-GO_BUILDMODE := -buildmode=default
+GO_BUILDMODE := -buildmode=pie
 GO_GCFLAGS :=
 GO_ASMFLAGS :=
 GO_MODFLAGS := $(if $(wildcard $(SOURCEDIR)/vendor/modules.txt),-mod=vendor,-mod=readonly)
